### PR TITLE
Build: publish artifacts to correct GCS buckets for main and release builds

### PR DESCRIPTION
**Pull request by malcolmholmes**
The publish location was changed for deb/rpm/etc packages for releases. However, this was also used within the main build, so main build packages were placed in the wrong place for the next "publish" step to find.

This fixes that - it provides the correct bucket to the `upload` step depending on whether we are in `main` or `release` phase.

### CODE REVIEW

**Review by torkelo — APPROVED** [review]

File: scripts/drone/steps/lib.star [hunk 1]
```diff
@@ -911,12 +916,13 @@ def upload_packages_step(edition, ver_mode, is_downstream=False):
...
-    packages_bucket = ' --packages-bucket $${PRERELEASE_BUCKET}/artifacts/downloads' + enterprise2_suffix(edition)
-
     if ver_mode == 'test-release':
         cmd = './bin/grabpl upload-packages --edition {} '.format(edition) + \
               '--packages-bucket grafana-downloads-test'
+    elif ver_mode == 'main':
+        cmd = './bin/grabpl upload-packages --edition {} --packages-bucket grafana-downloads'.format(edition)
     else:
+        packages_bucket = ' --packages-bucket $${PRERELEASE_BUCKET}/artifacts/downloads' + enterprise2_suffix(edition)
         cmd = './bin/grabpl upload-packages --edition {}{}'.format(edition, packages_bucket)
```
Comment (dsotirakis) [inline scripts/drone/steps/lib.star:926]: Do we need a space between {}{}? 🤔 
```suggestion
        cmd = './bin/grabpl upload-packages --edition {} {}'.format(edition, packages_bucket)
```

**Review by dsotirakis — APPROVED** [review]
One comment! 🚀

File: scripts/drone/steps/lib.star (hunk 1 above)
Comment (malcolmholmes) [reply scripts/drone/steps/lib.star:926]: No, the space is a few lines above (925) before the switch.

Comment (dsotirakis) [reply scripts/drone/steps/lib.star:926]: 🤦 Confusing! Go on then!

**Comment by marefr** [comment]
@malcolmholmes Merging this to main seems to have broken the build windows installer step, see [build](https://drone.grafana.net/grafana/grafana/43877/2/4)

**Pull request merged**